docs/websites/varnish/use-varnish-with-nginx-to-serve-cached-wordpress-content-over-ssl-on-debian-8.md

```diff
@@ -0,0 +1,683 @@
+
+	{: .file-excerpt }
+	/etc/varnish/custom.vcl
+	:   ~~~ conf
+	
+		sub vcl_deliver {
+		if (req.http.X-Purger) {
+		set resp.http.X-Purger = req.http.X-Purger;
+		  }
+		}
+		~~~
+
+	This concludes the `custom.vcl` configuration. You can save and exit the file. The final `custom.vcl` file will look like this:
+
+	{: .file }
+	/etc/varnish/custom.vcl
+	:   ~~~ conf
+	
+		vcl 4.0;
+
+		backend default {
+		.host = "localhost";
+		.port = "8080";
+		}
+
+		acl purger {
+		"localhost";
+		"your_server_ip";    
+		"your_server_ipv6";
+		}
+
+		sub vcl_recv {
+
+		 if (client.ip != "127.0.0.1" && req.http.host ~ "example-over-https.com") {
+			set req.http.x-redir = "https://www.example-over-https.com" + req.url;
+			return(synth(850, ""));
+		 }
+
+		 if (req.method == "PURGE") {
+			if (!client.ip ~ purger) {
+			return(synth(405, "This IP is not allowed to send PURGE requests."));
+		  }
+		 return (purge);
+		 }
+
+		 if (req.restarts == 0) {
+			if (req.http.X-Forwarded-For) {
+			set req.http.X-Forwarded-For = client.ip;
+			}
+		 }
+
+		 if (req.http.Authorization || req.method == "POST") {
+			return (pass);
+		 }
+
+		 if (req.url ~ "/feed") {
+			return (pass);
+		 }
+
+		 if (req.url ~ "wp-admin|wp-login") {
+			return (pass);
+		 }
+
+		 set req.http.cookie = regsuball(req.http.cookie, "wp-settings-\d+=[^;]+(; )?", "");
+		 set req.http.cookie = regsuball(req.http.cookie, "wp-settings-time-\d+=[^;]+(; )?", "");
+
+		 if (req.http.cookie == "") {
+		 unset req.http.cookie;
+		 }
+		}
+
+		sub vcl_synth {
+		 if (resp.status == 850) {  
+			 set resp.http.Location = req.http.x-redir;       
+			 set resp.status = 302;
+			 return (deliver);
+		 }
+		}
+
+		sub vcl_purge {
+		 set req.method = "GET";
+		 set req.http.X-Purger = "Purged";
+		 return (restart);
+		}
+
+		sub vcl_backend_response {
+		 set beresp.ttl = 24h;
+		 set beresp.grace = 1h;
+
+		 if (bereq.url !~ "wp-admin|wp-login|product|cart|checkout|my-account|/?remove_item=") {
+		 unset beresp.http.set-cookie;    
+		 }
+		}
+
+		sub vcl_deliver {
+		 if (req.http.X-Purger) {
+			set resp.http.X-Purger = req.http.X-Purger;
+		 }
+		}
+		~~~
+
+
+	For Varnish to work we also need to edit the `varnish.service` file to use our custom configuration file, with the port number and allocated memory values updated to match the changes we made in our `/etc/default/varnish` file.
+
+21. Open the `/lib/systemd/system/varnish.service` file:
+
+        sudo nano /lib/systemd/system/varnish.service
+
+22. Find the two lines beginning with  `ExecStart` and edit them to look like this:
+
+	{: .file-excerpt }
+	lib/systemd/system/varnish.service
+	:   ~~~ conf
+	
+		ExecStartPre=/usr/sbin/varnishd -C -f /etc/varnish/custom.vcl
+		ExecStart=/usr/sbin/varnishd -a :80 -T localhost:6082 -f /etc/varnish/custom.vcl -S /etc/varnish/secret -s malloc,500M
+		~~~
+
+23. After saving and exiting the file you have to reload the systemd process:
+
+        sudo systemctl daemon-reload
```

Comment:
After making any changes to a .service file in `/lib/systemd/system`, the systemd process has to be reloaded with the command: `sudo systemctl daemon-reload` . If we change any .service file in `/lib/systemd/system` but don't reload the systemd process, when trying to start that service we will receive a warning like:
`Warning: Unit file of ....service changed on disk, 'systemctl daemon-reload' recommended.`
And only after reloading the systemd process the new changes that we made will take effect. So, after following the guide up to this step, after installing and configuring Varnish as the guide describes, a user must reload the systemd process with `sudo systemctl daemon-reload` in order to have Varnish start with the new configuration. But Varnish shouldn't be started yet. Please note that at this point of the guide Varnish hasn't been started. Two lines below, the guide states: `We won't start Varnish at this point because we need to configure nginx first.` So, since nginx isn't configured yet, Varnish shouldn't be started. And of course Nginx should be started either, since it hasn't been configured yet.

The output that you posted above, shows that Varnsh is already running after the `sudo systemctl daemon-reload` command, on your test server, which is not as it should be. This means you started Varnish earlier during some tests. But at this point of the guide, step 23 of the 'Install and Configure Varnish' section , after installing and configuring Varnish, as I said, Varnish is not yet started. It will be started only after configuring  and starting Nginx, in step10 of the 'Configure nginx' section.

Also, if Varnish is running properly, when checking its status with `sudo systemctl status varnish`, the CGroup lines of the output should look like:
`6189 /usr/sbin/varnishd -a :80 -T localhost:6082 -f /etc/varnish...`
Your output shows `11606 /usr/sbin/varnishd -a :6081 -T localhost:6082 -f /etc/varnish/default....` The port on which Varnsh listens, as we just set it in `/lib/systemd/system/varnish.service` should be 80 not 6081.